toolz/dicttoolz.py

```diff
@@ -313,3 +313,21 @@ def get_in(keys, coll, default=None, no_default=False):
         if no_default:
             raise
         return default
+
+
+def select_keys(d, keys, factory=dict):
+    """ Select only certain keys from a dictionary
+
+    If supplied keys are not found in the dictionary, returns an empty
+    dictionary.
+
+    >>> d = {'a': 1, 'b': 2, 'c': 3}
+    >>> select_keys(d, ['a', 'b', 'd'])
+    {'a': 1, 'b': 2}
+    >>> select_keys(d, ['d'])
+    {}
+    """
+    rv = factory()
+    for k in set(keys).intersection(d.keys()):
```

Comment:
I recommend spelling this out explicitly for performance reasons:

```python
for k in keys:
    if k in d:
        rv[k] = d[k]
```

Reply:
Fixed.